sky/jobs/utils.py

```diff
@@ -1690,13 +1690,21 @@ def stream_logs(cls,
     def set_pending(cls, job_id: int, managed_job_dag: 'dag_lib.Dag',
                     workspace: str, entrypoint: str) -> str:
         dag_name = managed_job_dag.name
+        pool = managed_job_dag.pool
         # Add the managed job to queue table.
         code = textwrap.dedent(f"""\
             set_job_info_kwargs = {{'workspace': {workspace!r}}}
             if managed_job_version < 4:
                 set_job_info_kwargs = {{}}
             if managed_job_version >= 5:
                 set_job_info_kwargs['entrypoint'] = {entrypoint!r}
+            if managed_job_version >= 8:
+                from sky.serve import serve_state
+                pool_hash = None
+                if {pool!r} is not None:
```

Comment:
We will get this error `SyntaxWarning: "is not" with 'str' literal. Did you mean "!="?`
Use `{pool!r} != None` instead